backend/src/test/scala/com/softwaremill/adopttapir/starter/StarterServiceITTest.scala

```diff
@@ -6,20 +6,14 @@ import cats.effect.unsafe.implicits.global
 import cats.instances.list._
 import cats.syntax.parallel._
 import com.softwaremill.adopttapir.starter.api._
-import com.softwaremill.adopttapir.template.ProjectTemplate
-import com.softwaremill.adopttapir.test.BaseTest
+import com.softwaremill.adopttapir.test.ServiceTimeouts.waitForPortTimeout
+import com.softwaremill.adopttapir.test.{BaseTest, GeneratedService}
 import org.scalatest.{Assertions, ParallelTestExecution}
-import os.SubProcess
 import sttp.client3.{HttpURLConnectionBackend, Identity, SttpBackend, UriContext, asStringAlways, basicRequest}
 
-import java.io.{PrintWriter, StringWriter}
-import java.time.LocalDateTime
-import scala.annotation.tailrec
 import scala.collection.mutable
-import scala.concurrent.TimeoutException
 import scala.concurrent.duration.{DurationInt, FiniteDuration}
-import scala.util.{Properties, Using}
-import scala.util.matching.Regex
+import scala.util.Properties
 
 object Setup {
   lazy val jsonImplementations: List[JsonImplementationRequest] = {
```

Comment:
I propose to hide the variables which are not use externally. It helps in readability what is used outside. 

```
object Setup {
  type TestFunction = Integer => IO[Unit]

  val validConfigurations: Seq[StarterDetails] = for {
    effect <- EffectRequest.values
    server <- ServerImplementationRequest.values
    docs <- List(true, false)
    metrics <- List(true, false)
    json <- jsonImplementations
    scalaVersion <- scalaVersions
    builder <- BuilderRequest.values
    starterRequest = StarterRequest(
      "myproject",
      "com.softwaremill",
      effect,
      server,
      addDocumentation = docs,
      addMetrics = metrics,
      json,
      scalaVersion,
      builder
    )
    starterDetails <- FormValidator.validate(starterRequest).toSeq
  } yield starterDetails

  private lazy val jsonImplementations: List[JsonImplementationRequest] = {
    Properties
      .envOrElse("JSON", JsonImplementationRequest.values.mkString(","))
      .split(",")
      .map(JsonImplementationRequest.withName)
      .toList
  }

  private lazy val scalaVersions: List[ScalaVersionRequest] = {
    Properties.envOrElse("SCALA", ScalaVersionRequest.values.mkString(",")).split(",").map(ScalaVersionRequest.withName).toList
  }
}
```